### PR TITLE
Loshchilov2017decoupled+slycane9

### PR DESCRIPTION
**Pull request by slycane9**

### CODE REVIEW

**Comment by ishanashah** [comment]
I really like the review. It's very clear and concise, but also covers all the main ideas of the paper. The images are really helpful as well. There's only a couple small things I think could be improved:

1. Your description of Figure 2 can be improved. You said "Results show that decoupled weight decay allows the Adam optimizer to compete with SGD in image classification in terms of having a separable hyperparameter space." Figure 2 shows the effect of decoupling weight decay for both Adam and SGD, not just ADAM. The idea is that the left shows SGD and ADAM when L2 regularization is not decoupled from learning rate, and the right shows SGD and ADAM when it is decoupled. The charts on the left are diagonal, showing that when the hyperparameters are coupled you have to change one whenever you change another, making it very difficult to tune hyperparameters. The right hand side shows that with decoupled weight decay, both SGD and ADAM benefit because each hyperparameter can be tuned independently (there is no diagonal). A more accurate description would be "Results show that for both SGD and ADAM, decoupling weight decay makes the weight decay and learning rate hyperparameters independent."

2. I'm not sure if this matter's but it could be nice to have a line about implementation in pytorch. You can probably link this:
https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html in case someone wants to try using it.
